### PR TITLE
add JS resolver support

### PR DESCRIPTION
**Pull request by iartemiev**
Originally opened [from fork branch](https://github.com/aws-amplify/amplify-backend/pull/1073). Re-opening from upstream branch to run e2es.

## Changes

The Amplify CDK construct does not currently support provisioning AppSync JS resolvers for fields defined in the customer's Amplify schema.

This PR leverages the AppSync L1 CDK construct escape hatch to add support. 

Related data-schema PR: https://github.com/aws-amplify/amplify-api-next/pull/113

**Corresponding docs PR, if applicable:** https://github.com/aws-amplify/docs/pull/6836/files#diff-dfe0a992e880882948237e578edffdfb4a4548f19487a58118ff900d2c7c233aR178-R210

### Customer DX

```ts
const schema = a.schema({
  Post: a.model({...}),

  likePost: a.mutation()
    .returns( a.string() )
    .authorization([ a.allow.private() ])
    .handler([
      // a.handler.custom is how customers define an AppSync JS Resolver
      a.handler.custom({
        entry: './myJsResolver.js', // relative or absolute path to file containing AppSync JS resolver function
        dataSource: a.ref('Post') // optional; can reference existing model or string data source; defaults to 'NONE_DS'
      })
    ])
})

export const data = defineData({
  schema,
});
```

This definition returns the following IR from `data-schema` when calling `schema.transform()`:
```ts
{
  typeName: 'Mutation',
  fieldName: 'likePost',
  handlers: [
    {
      dataSource: 'PostTable',
      entry: { relativePath: './myJsResolver.js', importLine: `${stack_trace_import_line}` }
    }
  ]
}
```

The`convertJsResolverDefinition`function added in this PR consumes the IR and provisions an AppSync JS Pipeline Resolver and a list of functions.

## Checklist

- [x] If this PR includes a functional change to the runtime behavior of the code, I have added or updated automated test coverage for this change.
- [ ] If this PR requires a change to the [Project Architecture README](../PROJECT_ARCHITECTURE.md), I have included that update in this PR.
- [x] If this PR requires a docs update, I have linked to that docs PR above.
- [x] If this PR modifies E2E tests, makes changes to resource provisioning, or makes SDK calls, I have run the PR checks with the `run-e2e` label set.

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

**Review by edwardfoyle — APPROVED** [review]

File: packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts [hunk 1]
```diff
@@ -174,8 +174,8 @@ class DataStorageAuthWithTriggerTestProject extends TestProjectBase {
         sourceFile: sourceDataResourceFile,
         projectFile: dataResourceFile,
         deployThresholdSec: {
-          onWindows: 40,
-          onOther: 30,
+          onWindows: 135,
+          onOther: 125,
```
Comment (sobolk) [inline packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts:178]: I left a comment here https://github.com/aws-amplify/amplify-backend/pull/1089#discussion_r1508380311 .
What do you think?

Comment (iartemiev) [reply packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts:178]: Sorry, I missed that. Happy to add a separate test app to make this easier to track. 
Can we merge this as-is and I'll follow up with that in another PR later today or Monday?

Comment (sobolk) [reply packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts:178]: @edwardfoyle or the other way round. I.e. have data-with-trigger-and-other-stuff use minimal schema and have this new test explore complex schema(s) .

Comment (sobolk) [reply packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts:178]: > Can we merge this as-is and I'll follow up with that in another PR later today or Monday?

yes. if you can add `release-blocker` GH issue with `data-team` label on it and commit an ECD to solve it (we're at times where we want to stop adding debt).

Comment (iartemiev) [reply packages/integration-tests/src/test-project-setup/data_storage_auth_with_triggers.ts:178]: Created: https://github.com/aws-amplify/amplify-backend/issues/1097

Will start work on it ASAP. Let's shoot for 3/4 ECD

**Review by sobolk — APPROVED** [review]

**Pull request merged**